main.go

```diff
@@ -82,19 +82,37 @@ func main() {
 		log.Fatalf(usage)
 	}
 
-	url, err := url.Parse(args[0])
+	uri := schemify(args[0])
+
+	url, err := url.Parse(uri)
 	if err != nil {
-		log.Fatalf("could not parse url %q: %v", args[0], err)
+		log.Fatalf("could not parse url %q: %v", uri, err)
 	}
 
 	visit(url)
 }
 
+func schemify(uri string) string {
+	if strings.Contains(uri, "://") != true {
+		if strings.HasSuffix(uri, ":80") != true {
+			return "https://" + uri
+		}
+		return "http://" + uri
+	}
+	return uri
+}
+
 // visit visits a url and times the interaction.
 // If the response is a 30x, visit follows the redirect.
 func visit(url *url.URL) {
 	scheme := url.Scheme
 	hostport := url.Host
+
+	// No hostname, just a port
+	if strings.HasPrefix(hostport, ":") {
```

Comment:
Please move this inside the function at 116. The function at 116 should also be broken out into its own function as it now contains a lot of business logic.